### DISCUSSION TITLE
what can be the solutions to get the original image back from the image pyramids present at one level

### DISCUSSION THREAD

**Discussion by ghost** [libvips/libvips#2835] [Q&A]
i created image pyramids using "vips dzsave image.png mydz --suffix .png" and then tried to combine the image tiles at one zoom level using imagemagic command "convert +append image_*.png output.png" but its giving the distorted image (image tiles are not added in correct order).why this is happening?

**Comment by jcupitt** [comment]
Hi @amit-shew,

At a guess, because the `*.png` will sort images lexicographically and not numerically.

You can split and join images like this:

```
$ vips dzsave wtc.jpg x --overlap 0
$ cd x_files/14
$ vips arrayjoin "$(ls *.jpeg | sort -t_ -k2g -k1g)" x.jpg --across 37
```

Where `wtc.jpg` is a 10k x 10k pixel RGB image.

The `sort` does a numerical sort by first y and then x of filenames of the form `xxx_yyy.jpeg`.

**Comment by ghost** [comment]


**Reply by jcupitt** [reply]
You need to show the exact commands you ran so I can reproduce your problem.

**Reply by jcupitt** [reply]
Eg. with your sample image:

```
$ vips dzsave apple.jpeg x --overlap 0
$ cd x_files/9
$ ls
0_0.jpeg  0_1.jpeg  1_0.jpeg  1_1.jpeg
$ vips arrayjoin "$(ls *.jpeg | sort -t_ -k2g -k1g)" x.jpg --across 2
```

Makes:

![x](https://user-images.githubusercontent.com/580843/171855486-53bccbb5-3910-480b-b510-b7e085a5242b.jpg)

**Comment by ghost** [comment]
thanks @jcupitt
